internal/controller/postgrescluster/pgbackrest.go

```diff
@@ -561,32 +560,35 @@ func (r *Reconciler) generateRepoHostIntent(postgresCluster *v1beta1.PostgresClu
 		repo.Spec.Replicas = initialize.Int32(1)
 	}
 
+	// Restart containers any time they stop, die, are killed, etc.
+	// - https://docs.k8s.io/concepts/workloads/pods/pod-lifecycle/#restart-policy
+	repo.Spec.Template.Spec.RestartPolicy = corev1.RestartPolicyAlways
+
+	// The pgBackRest TLS server handles client connections in detached/orphaned
```

Comment:
:pick:  I understand that there are cases where we'll have an orphaned child process, so we want to use the Kubernetes pause process to reap them, but I'm missing how the pause process is actually being used. More clarification here might be helpful.

Reply:
Setting `spec.shareProcessNamespace: true` is what sets the pause process to PID 1.

My only idea is to say that in the comment, but that's also what the line of code says. Any ideas how to phrase it?

Reply:
Looking at this again, I think my confusion was that the `shareProcessNamespace` setting BOTH makes processes visible across containers in a Pod AND results in the PID 1 `/pause` process.

I think adding another link would make this clearer, but then it would be repetitive, i.e.
```
// The pgBackRest TLS server handles client connections in detached/orphaned
// child processes. Use Kubernetes' pause process as PID 1 to reap them.
// - https://github.com/pgbackrest/pgbackrest/blob/release/2.36/src/command/server/server.c#L51
// - https://kubernetes.io/docs/tasks/configure-pod-container/share-process-namespace/#understanding-process-namespace-sharing
//
// The pgBackRest TLS server must be signaled when its configuration or
// certificates change. Let containers see each other's processes.
// - https://docs.k8s.io/tasks/configure-pod-container/share-process-namespace/
```
So I'm not sure that's the way to go. It may not be confusing to others, so perhaps it's fine as is.

Reply:
Now that I read those links again, they don't mention child processes or reaping. I see if I can add some language around that (undocumented?) aspect of the pause process.

Reply:
The updated comment and the additional link really help :100: